contracts/test/expanders-test.ts

```diff
@@ -507,4 +508,168 @@ describe("Expanders", async function () {
+                - 1: Use registry for BLS key
+                - 1: Include a tx.origin payment
+                - 1: Use registry for ERC20 address
+                - 1: Use registry for recipient address
+      000000  - Registry index for sendWallet's public key
+      00      - nonce: 0
+      0bda28  - gas: 92,483
+      02      - two actions
+      000000  - Registry index for ERC20 address
+      00      - transfer
+      000002  - Registry index for recipient address
+      9100    - 0.1 MCK
+      6d00    - Pay 0.000005 ETH to tx.origin
+
+      279ef4bad05ee777d6c6c497592676242f51a9150f6b7c0e4ae5b36445c6d248
+      0613346680c8f2b9c3a143a378c2e56a770da56b76d9cb0b8eee29a2e2ac87b7
+              - signature
+    */
+
+    const compressedBundle = await fx.bundleCompressor.compress(bundle);
+
+    await receiptOf(fx.blsExpanderDelegator.run(compressedBundle));
+
+    await expect(erc20.balanceOf(sendWallet.address)).to.eventually.eq(
+      ethers.utils.parseEther("0.9"),
+    );
+
+    await expect(erc20.balanceOf(recvWallet.address)).to.eventually.eq(
+      ethers.utils.parseEther("1.1"),
+    );
+  });
+
+  it("should approve max ERC20 using registries", async function () {
+    const fx = await Fixture.getSingleton();
+    const { addressRegistry, blsPublicKeyRegistry } = fx.fallbackCompressor;
+    const [sendWallet, recvWallet] = await fx.createBLSWallets(2);
+    const singletonFactory = await SafeSingletonFactory.init(fx.signers[0]);
+
+    const erc20 = await singletonFactory.connectOrDeploy(MockERC20Factory, [
+      "MockToken",
+      "MCK",
+      0,
+    ]);
+
+    await Promise.all([
+      addressRegistry.register(erc20.address),
+      ...[sendWallet, recvWallet]
+        .map((wallet) => [
+          blsPublicKeyRegistry.register(wallet.PublicKey()),
+          addressRegistry.register(wallet.address),
+        ])
+        .flat(),
+    ]);
+
+    const bundle = await sendWallet.signWithGasEstimate({
+      nonce: await sendWallet.Nonce(),
+      actions: [
+        {
+          ethValue: 0,
+          contractAddress: erc20.address,
+          encodedFunction: erc20.interface.encodeFunctionData("approve", [
+            recvWallet.address,
+            ethers.constants.MaxUint256,
+          ]),
+        },
+      ],
+    });
+
+    /*
+      Example:
+
+      01      - One operation
+      02      - Use expander 2 (ERC20 expander)
+      0d      - 0x0d = 0b1101 bit stream:
+                - 1: Use registry for BLS key
+                - 0: Do not include a tx.origin payment
+                - 1: Use registry for ERC20 address
+                - 1: Use registry for spender address
+                (The read order is 'backwards' because the lowest bit is read
+                first)
+      000000  - Registry index for sendWallet's public key
+      00      - nonce: 0
+      0bb917  - gas: 58,555
+      01      - one action
+      000000  - Registry index for ERC20 address
+      03      - approve (max)
```

Comment:
Should we add a test for `approve`, `transferFrom` and `mint`?

Reply:
Probably :)